### PR TITLE
[ADD] Thailand localisation Documentation

### PR DESCRIPTION
**Pull request by tap-odoo**
Added Thailand localisation documentation as it not exist before. Features were pushed to Odoo 16 already

### CODE REVIEW

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+generate a CSV file for bills to upload to the `RDprep for Thailand e-Filling application <https://efiling.rd.go.th/rd-cms/>`_.
+
+To generate a PND CSV file, select a specific time or time range on the tax report,
+and click :guilabel:`PND3` or :guilabel:`PND53`.
+
+This generates the :file:`Tax Report PND3.csv` and :file:`Tax Report PND53.csv`
+file that lists all the vendor bill lines with the applicable withholding tax.
+
+  .. image:: thailand/thailand-pnd3-pnd53.png
+     :align: center
```
Comment (tong-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:66]:
```suggestion
.. image:: thailand/thailand-pnd3-pnd53.png
   :align: center
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+Tax Invoice
+===========
+
+The Tax Invoice PDF report can be generated from Odoo on Invoicing module. User has the option to
+print PDF reports for normal invoices and tax invoices. To print out Tax Invoices, users can click
+on :guilabel:`Print Invoices` in Odoo. Normal invoices can be printed as
+Commercial Invoice by clicking on :menuselection:`⚙️ --> Print --> Commercial Invoice`.
+
+  .. image:: thailand/thailand-tax-invoice.png
+     :align: center
```
Comment (tong-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:82]:
```suggestion
.. image:: thailand/thailand-tax-invoice.png
   :align: center
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Sales and Purchase Tax Report
+-----------------------------
+
+To generate a Sales and Purchase Tax report, select a specific time or time range on the tax report,
+and click on :guilabel:`VAT-202-01 (xlsx)` for Purchase tax
+and :guilabel:`VAT-202-02 (xlsx)` for Sales tax.
+
+  .. image:: thailand/thailand-tax-report.png
+     :align: center
```
Comment (tong-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:40]:
```suggestion
.. image:: thailand/thailand-tax-report.png
   :align: center
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+Headquarter/ Branch number settings
+-----------------------------------
+
+Setting is done in Contact Application.
+This information can be set on :menuselection:`Contact --> under Sales & Purchase tab --> Company ID`.
+Leave this field blank if the contact is **Headquarters** and input the **Branch number**
+if the contact is identified as a branch.
+  .. image:: thailand/thailand-contact.png
+     :align: center
```
Comment (tong-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:92]:
```suggestion
if the contact is identified as a branch.

.. image:: thailand/thailand-contact.png
   :align: center
```

**Comment by vin-odoo** [comment]
@odoo/localizations-doc-review Hi team :pray: Could you please review this PR?
It would be perfect if it could be merged and available before September so that it can be shared during a webinar.

Sorry for the team ping and thank you in advance :pray:

**Review by toaa-odoo — CHANGES_REQUESTED** [review]
Hi, thank you for your great work!

Since you created it on your own repository, I couldn't push my commit on it. Instead, you can copy/paste this into your branch (which is my changes and recommendations in RST). Note there is also an (already compressed) new image to add in the configuration section:

========
Thailand
========

Configuration
=============

:ref:`Install <general/install>` the :guilabel:`🇹🇭 Thailand` localization package to get all the
features of the Thai localization:

.. list-table::
   :header-rows: 1

   * - Name
     - Technical name
     - Description
   * - :guilabel:`Thailand - Accounting`
     - `l10n_th`
     - Default :ref:`fiscal localization package <fiscal_localizations/packages>`
   * - :guilabel:`Thailand - Accounting Reports`
     - `l10n_th_reports`
     - Country-specific accounting reports

.. image:: thailand/thailand-modules.png
   :alt: Thailand localization modules

Chart of accounts and taxes
===========================

Odoo’s fiscal localization package for Thailand includes the following taxes:

- VAT 7%
- VAT-exempted
- Withholding tax
- Withholding income tax

Tax report
==========

Odoo allows users to generate Excel files to submit their VAT to the **Revenue Department** of
Thailand.

Sales and purchase tax report
-----------------------------

To generate a sales and purchase tax report, go to :menuselection:`Accounting --> Reporting --> Tax
Report`. Select a specific time or time range on the tax report, and click
:guilabel:`VAT-202-01 (xlsx)` for purchase tax, and :guilabel:`VAT-202-02 (xlsx)` for sales tax.

.. image:: thailand/thailand-tax-report.png
   :alt: Thai purchase and sales taxes reports

Withholding PND tax report
--------------------------

PND report data displays the summarized amounts of the applicable **withholding corporate income
tax returns (domestic)** from vendor bills under the :guilabel:`PND53 (TH)` and
:guilabel:`PND3 (TH)` tax reports, and is installed by default with the Thai localization.

.. image:: thailand/thailand-pnd-report.png
   :alt: PND tax reports

.. note::
   Withholding corporate income tax returns (domestic) is the tax used in case the company has
   withheld the tax from “**Personal (PND3)**” or “**Corporate (PND53)**” services provided such as
   rental, hiring, transportation, insurance, management fee, consulting, and etc.

The PND Tax report allows users to generate a CSV file for bills to upload on the
`RDprep for Thailand e-Filling application <https://efiling.rd.go.th/rd-cms/>`_.

To generate a PND CSV file, go to :menuselection:`Accounting --> Reporting --> Tax Report`, select a
specific time or time range on the tax report, and click :guilabel:`PND3` or :guilabel:`PND53`.

This generates the :file:`Tax Report PND3.csv` and :file:`Tax Report PND53.csv` file that lists all
the vendor bill lines with the applicable withholding tax.

.. image:: thailand/thailand-pnd3-pnd53.png
   :alt: PND3 and PND53 CSV files

.. warning::
   Odoo cannot generate the PND or PDF report or **withholding tax certificate** directly. The
   generated :file:`Tax Report PND3.csv` and :file:`Tax Report PND53.csv` files must be exported
   to an external tool to convert it into a **withholding PND** report or **PDF** format.

Tax invoice
===========

The **tax invoice PDF** report can be generated from Odoo through the **Invoicing** module. Users
have the  option to print PDF reports for normal invoices and tax invoices. To print out
**tax invoices**, users can click on :guilabel:`Print Invoices` in Odoo. Regular invoices can be
printed as **commercial invoice** by clicking on :menuselection:`⚙️ (Cog) --> Print --> Commercial
Invoice`.

.. image:: thailand/thailand-tax-invoice.png
   :alt: Commercial invoice printing

Headquarter/Branch number settings
-----------------------------------

You can inform the **Headquarters** and **Branch number** of a company in the **Contacts** app. Once
in the app, open the **contact form** of the company and under the :guilabel:`Sales & Purchase` tab:

- If the contact is identified as a branch, input the **Branch number** in the
  :guilabel:`Company ID` field.
- If the contact is a **Headquarters**, leave the :guilabel:`Company ID` field **blank**.

.. image:: thailand/thailand-contact.png
   :alt: Company Headquarter/Branch number

.. tip::
   This information is used in the **tax invoice** PDF report and PND **tax report** export.

thailand-modules

**Comment by tong-odoo** [comment]
@toaa-odoo Since this PR is quite urgent therefore I will help updating the PR. May I ask if we could finish the update within this week. Is it possible to deploy this to production before 7 Sep?

**Comment by toaa-odoo** [comment]
> @toaa-odoo Since this PR is quite urgent therefore I will help updating the PR. May I ask if we could finish the update within this week. Is it possible to deploy this to production before 7 Sep?

Passed on the message to somebody with R+ rights

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+   withheld the tax from “**Personal (PND3)**” or “**Corporate (PND53)**” services provided such as
+   rental, hiring, transportation, insurance, management fee, consulting, and etc.
+
+The PND Tax report allows users to generate a CSV file for bills to upload on the
+`RDprep for Thailand e-Filling application <https://efiling.rd.go.th/rd-cms/>`_.
+
+To generate a PND CSV file, go to :menuselection:`Accounting --> Reporting --> Tax Report`, select a
+specific time or time range on the tax report, and click :guilabel:`PND3` or :guilabel:`PND53`.
+
+This generates the :file:`Tax Report PND3.csv` and :file:`Tax Report PND53.csv` file that lists all
```
Comment (xpl-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:74]:
```suggestion
This generates the :file:`Tax Report PND3.csv` and :file:`Tax Report PND53.csv` files that list all
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+.. image:: thailand/thailand-pnd-report.png
+   :alt: PND tax reports
+
+.. note::
+   Withholding corporate income tax returns (domestic) is the tax used in case the company has
+   withheld the tax from “**Personal (PND3)**” or “**Corporate (PND53)**” services provided such as
+   rental, hiring, transportation, insurance, management fee, consulting, and etc.
+
+The PND Tax report allows users to generate a CSV file for bills to upload on the
```
Comment (xpl-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:68]:
```suggestion
The PND tax report allows users to generate a CSV file for bills to upload on the
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+This generates the :file:`Tax Report PND3.csv` and :file:`Tax Report PND53.csv` file that lists all
+the vendor bill lines with the applicable withholding tax.
+
+.. image:: thailand/thailand-pnd3-pnd53.png
+   :alt: PND3 and PND53 CSV files
+
+.. warning::
+   Odoo cannot generate the PND or PDF report or **withholding tax certificate** directly. The
+   generated :file:`Tax Report PND3.csv` and :file:`Tax Report PND53.csv` files must be exported
+   to an external tool to convert it into a **withholding PND** report or **PDF** format.
```
Comment (xpl-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:83]:
```suggestion
   to an external tool to convert them into a **withholding PND** report or a **PDF** file.
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+Tax invoice
+===========
+
+The **tax invoice PDF** report can be generated from Odoo through the **Invoicing** module. Users
+have the  option to print PDF reports for normal invoices and tax invoices. To print out
+**tax invoices**, users can click on :guilabel:`Print Invoices` in Odoo. Regular invoices can be
+printed as **commercial invoice** by clicking on :menuselection:`⚙️ (Cog) --> Print --> Commercial
```
Comment (xpl-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:91]:
```suggestion
printed as **commercial invoices** by clicking on :menuselection:`Cog button (⚙️) --> Print --> Commercial
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+Invoice`.
+
+.. image:: thailand/thailand-tax-invoice.png
+   :alt: Commercial invoice printing
+
+Headquarter/Branch number settings
+----------------------------------
+
+You can inform the **Headquarters** and **Branch number** of a company in the **Contacts** app. Once
```
Comment (xpl-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:100]:
```suggestion
You can inform a company's **Headquarters** and **Branch number** in the **Contacts** app. Once
```

**Review by xpl-odoo — COMMENTED** [review]
Very minor spelling suggestions. I haven't tested the described flow, though.

**Comment by tong-odoo** [comment]
> Very minor spelling suggestions. I haven't tested the described flow, though.

@xpl-odoo Thanks for the review! I updated the spelling mistake. Please feel free to let me know once you tested the described flow. Thanks again!

**Comment by xpl-odoo** [comment]
> > Very minor spelling suggestions. I haven't tested the described flow, though.
> 
> @xpl-odoo Thanks for the review! I updated the spelling mistake. Please feel free to let me know once you tested the described flow. Thanks again! 

@toaa-odoo did you test the flow? If yes, the PR should be ready to be merged. If not, could you? :)

**Comment by toaa-odoo** [comment]
> > > Very minor spelling suggestions. I haven't tested the described flow, though.
> > 
> > 
> > @xpl-odoo Thanks for the review! I updated the spelling mistake. Please feel free to let me know once you tested the described flow. Thanks again!
> 
> @toaa-odoo did you test the flow? If yes, the PR should be ready to be merged. If not, could you? :)

It was, you can r+ it

**Comment by xpl-odoo** [comment]
@robodoo r+

**Pull request closed without merging**